provisioning/miniprov/hare_mp/store.py

```diff
@@ -110,6 +116,21 @@ def get_hostnames_for_service(self, service_type: str) -> List[str]:
             nodes.append(self.get(f'node>{machine_id}>hostname'))
         return nodes
 
+    def get_machine_ids_for_component(self, name: str) -> List[str]:
```

Comment:
Sorry, what is the exact difference between this function and "get_machine_ids_for_service"?
I think its confusing.
Can we have only one function?

Reply:
**Moved** the functionality into one single function - `get_machine_ids_for_attribute()`
Kept it as a wrapper - ` get_machine_ids_for_service() , get_machine_ids_for_component()` since the attribute type (parent and child) can be confusing, as discussed.
Thanks!

Reply:
done